### PR TITLE
[MediaAccessibility] Add xcode 15 beta 7 support.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/mediaaccessibility.cs [hunk 1]
```diff
@@ -45,4 +46,29 @@ interface MAVideoAccommodations {
...
+
+	[TV (17,0), Mac (14,0), iOS (17,0), MacCatalyst (17,0)]
+	[BaseType (typeof(NSObject))]
+	interface MAFlashingLightsProcessor
+	{
+		[Export ("canProcessSurface:")]
+		bool CanProcess (Surface surface);
+
+		[Export ("processSurface:outSurface:timestamp:options:")]
+		MAFlashingLightsProcessorResult Process (Surface inSurface, Surface outSurface, double timestamp, [NullAllowed] NSDictionary<NSString, NSObject> options);
```
Comment (rolfbjarne) [inline src/mediaaccessibility.cs:72]: Can the options be a strong dictionary?

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/mediaaccessibility.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/mediaaccessibility.cs:70]: I did not find any documentation about the keys or values. https://developer.apple.com/documentation/mediaaccessibility/maflashinglightsprocessoroptionkey?language=objc

Or in the header: https://github.com/xamarin/xamarin-macios/wiki/MediaAccessibility-iOS-xcode15.0-b1

**Review by haritha-mohan — APPROVED** [review]

File: src/mediaaccessibility.cs [hunk 2]
```diff
@@ -45,4 +46,27 @@ interface MAVideoAccommodations {
...
+	}
+
+	[TV (17, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (NSObject))]
+	interface MAFlashingLightsProcessor {
+		[Export ("canProcessSurface:")]
+		bool CanProcess (Surface surface);
+
+		[Export ("processSurface:outSurface:timestamp:options:")]
+		MAFlashingLightsProcessorResult Process (Surface inSurface, Surface outSurface, double timestamp, [NullAllowed] NSDictionary<NSString, NSObject> options);
```
Comment (dalexsoto) [inline src/mediaaccessibility.cs:70]: Probably this is better so we can strong'ify this in the future

```suggestion
		MAFlashingLightsProcessorResult Process (Surface inSurface, Surface outSurface, double timestamp, [NullAllowed] NSDictionary options);
```

**Review by dalexsoto — APPROVED** [review]

File: src/mediaaccessibility.cs (hunk 1 above)
Comment (stephen-hawley) [reply src/mediaaccessibility.cs]: FYI - I also looked at the Swift doc'n and the key, oddly enough, is *not* an `NSString`, which is an inner struct called `MAFlashingLightsProcessor.OptionKey` which has initializers from `String` - it feels like they're keeping their options open for the future, but the values of the options are also untyped there and undocumented.

**Review by rolfbjarne — APPROVED** [review]

File: src/mediaaccessibility.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/mediaaccessibility.cs]: I did notice the same, we should leave it as open as possible.

**Pull request merged**